packages/yarnpkg-core/sources/Configuration.ts

```diff
@@ -1038,9 +1038,18 @@ export class Configuration {
           continue;
 
         for (const userPluginEntry of data.plugins) {
-          const userProvidedPath = typeof userPluginEntry !== `string`
-            ? userPluginEntry.path
-            : userPluginEntry;
+          let userProvidedPath: string
+          if (typeof userPluginEntry !== `string`) {
+            if (userPluginEntry.enabled === `false`) {
```

Comment:
Note that this uses `\`false\`` as a string. This is because we do not have all the rcFile parsed at this time. Not sure how to do this better, if you do, please let me know.